src/RootCloseWrapper.js

```diff
@@ -1,20 +1,89 @@
-import contains from 'dom-helpers/query/contains'
-import listen from 'dom-helpers/events/listen'
-import PropTypes from 'prop-types'
-import React from 'react'
-import ReactDOM from 'react-dom'
+import contains from 'dom-helpers/query/contains';
+import listen from 'dom-helpers/events/listen';
+import PropTypes from 'prop-types';
+import React, { useEffect, useRef } from 'react';
+import ReactDOM from 'react-dom';
+import useEventCallback from '@restart/hooks/useEventCallback';
+import warning from 'warning';
 
-import ownerDocument from './utils/ownerDocument'
+import ownerDocument from './utils/ownerDocument';
 
-const escapeKeyCode = 27
-const noop = () => {}
+const escapeKeyCode = 27;
+const noop = () => {};
 
 function isLeftClickEvent(event) {
-  return event.button === 0
+  return event.button === 0;
 }
 
 function isModifiedEvent(event) {
-  return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey)
+  return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
+}
+
+export function useRootClose(
+  ref,
+  onRootClose,
+  { disabled, clickTrigger = 'click' } = {},
+) {
+  const preventMouseRootCloseRef = useRef(false);
+  const onClose = useEventCallback(onRootClose || noop);
+
+  const handleMouseCapture = e => {
+    const currentTarget = ref && ('current' in ref ? ref.current : ref);
+    warning(
+      !!currentTarget,
+      'RootClose captured a close event but does not have a ref to compare it to. ' +
```

Comment:
i'd just pre-concat these; now that we're on prettier, we won't get a line length lint hit anyway